### PR TITLE
Browserstack wdio service code

### PR DESCRIPTION
**Pull request by Ankit098**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,82 @@
...
+In order to authorize to the BrowserStack service your config needs to contain a [`user`](https://webdriver.io/docs/options#user) and [`key`](https://webdriver.io/docs/options#key) option.
+
+### browserstackLocal
+Set this to true to enable routing connections from Browserstack cloud through your computer. You will also need to set `browserstack.local` to true in browser capabilities.
+
+Type: `Boolean`<br />
+Default: `false`
+
+### preferScenarioName
+Cucumber only. Set this to true to enable updating the session name to the Scenario name if only a single Scenario was ran. Useful when running in parallel with [wdio-cucumber-parallel-execution](https://github.com/SimitTomar/wdio-cucumber-parallel-execution).
```
Comment (karanshah-browserstack) [inline README.md:59]: The default should be True? What will be the ScenarioName if it will be false?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  "dependencies": {
+    "@types/node": "^17.0.4",
+    "@wdio/logger": "7.16.0",
+    "@wdio/types": "7.16.14",
+    "browserstack-local": "^1.4.5",
+    "got": "^11.0.2",
+    "webdriverio": "7.16.15"
+  },
+  "devDependencies": {
+    "@tsconfig/node12": "^1.0.9"
```
Comment (karanshah-browserstack) [inline package.json]: Why is node12 in devDependencies? If we are using this config in our code, then we should include it in our dependencies else when npm release the package and user tries to install it using `npm i` command, this package doesn't get installed on it's own.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "baseUrl": ".",
+    "outDir": "./build",
+    "rootDir": "./src",
+    "inlineSourceMap": true
+  },
+  "include": [
+    "src/**/*",
+    "../../@types"
+  ]
+}
```
Comment (francisf) [inline tsconfig.json]:
```suggestion
}

```

File: package.json (hunk 2 above)
Comment (karanshah-browserstack) [reply package.json]: jest seems to be missing in devDependencies

File: src/launcher.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import { BrowserstackConfig } from './types'
+
+const log = logger('@browserstack/wdio-browserstack-service')
+
+type BrowserstackLocal = BrowserstackLocalLauncher.Local & {
+    pid?: number;
+    stop(callback: (err?: any) => void): void;
+}
+
+export default class BrowserstackLauncherService implements Services.ServiceInstance {
```
Comment (karanshah-browserstack) [inline src/launcher.ts:17]: Can we rename BrowserstackLauncherService to BrowserstackLocalService? Or add comments to intend the functionality of this file?

Comment (Ankit098) [inline src/launcher.ts:17]: Renaming would require changes in multiple other places as well, didn't change the naming convention already being used by the existing service. Will try to add some comments about the changes made.

File: package.json (hunk 2 above)
Comment (Ankit098) [reply package.json]: yes, will add jest needed for tests

File: README.md (hunk 1 above)
Comment (Ankit098) [reply README.md:59]: gets called with name of feature, if set to false.

File: LICENSE-MIT [hunk 5]
```diff
@@ -0,0 +1,20 @@
+Copyright (c) OpenJS Foundation and other contributors
```
Comment (karanshah-browserstack) [inline LICENSE-MIT]: Do we need this?

Comment (Ankit098) [reply LICENSE-MIT]: removed

**Review by karanshah-browserstack — DISMISSED** [review]

**Review by karanshah-browserstack — APPROVED** [review]

**Review by francisf — APPROVED** [review]

**Pull request merged**